File: src/FizzBuzz Variety Sample Codes/fizzbuzz.dart
// Fizzかどうかを判定する関数 --- (*1)
isFizz(int i) { return i % 3 == 0; }
// Buzzかどうかを判定する関数
isBuzz(int i) { return i % 5 == 0; }
// FizzBuzzに応じた値を返す --- (*2)
fizzbuzz(int i) {
  if (isFizz(i) && isBuzz(i)) return "FizzBuzz";
  if (isFizz(i)) return "Fizz";
  if (isBuzz(i)) return "Buzz";
  return i;
}
// 最初に実行する関数 --- (*3)
main() {
  // 1から100まで繰り返す --- (*4)
  for (var i = 1; i <= 100; i++) {
    var result = fizzbuzz(i);
    print(result);
  }
}

File: src/FizzBuzz Variety Sample Codes/fizzbuzz.rs
// プログラムはmain関数から始まる --- (*1)
fn main() {
    for i in 1 .. 100+1 {
        let result = fizzbuzz(i);
        println!("{0}", result)
    }
}
// FizzBuzzを返す関数を定義 --- (*2)
fn fizzbuzz(i:i32) -> String {
    // 無名関数を定義 --- (*3)
    let is_fizz = |i| { i % 3 == 0 };
    let is_buzz = |i| { i % 5 == 0 };
    // 順次判定していく --- (*4)
    if is_fizz(i) && is_buzz(i) {
        return "FizzBuzz".to_string();
    } else if is_fizz(i) {
        return "Fizz".to_string();
    } else if is_buzz(i) {
        return "Buzz".to_string();
    } else {
        return i.to_string();
    }
}
